### DISCUSSION TITLE
Updatе Golang dependencies error

### DISCUSSION THREAD

**Discussion by maymaewa** [renovatebot/renovate#19224] [Request Help]
Hello! I I found strange behavior when renovate updated any dependencies in **Golang** projects.
### Info: 
Renovate is self-hosted. We used **33** Renovate version and it's worked good. There was a problem starting **34** version. Bot updated dependency version in go.mod file and try to run "go tidy" command for updating go.sum file. After this i got next logs:

`DEBUG: Datasource connection error (repository=Company/libs/go/postgres-client, branch=renovate/go-libs)
       "datasource": "golang-version",
       "packageName": "golang",
       "url": undefined,
       "errCode": "ECONNRESET"
DEBUG: Failed to update go.sum (repository=Company/libs/go/postgres-client, branch=renovate/go-libs)
       "err": {
         "message": "No tool releases found.",
         "stack": "Error: No tool releases found.\n    at resolveConstraint (/usr/src/app/node_modules/renovate/lib/util/exec/containerbase.ts:205:11)\n    at generateInstallCommands (/usr/src/app/node_modules/renovate/lib/util/exec/containerbase.ts:256:27)\n    at prepareRawExec (/usr/src/app/node_modules/renovate/lib/util/exec/index.ts:145:11)\n    at exec (/usr/src/app/node_modules/renovate/lib/util/exec/index.ts:172:39)\n    at updateArtifacts (/usr/src/app/node_modules/renovate/lib/modules/manager/gomod/artifacts.ts:332:5)\n    at getUpdatedPackageFiles (/usr/src/app/node_modules/renovate/lib/workers/repository/update/branch/get-updated.ts:252:23)\n    at processBranch (/usr/src/app/node_modules/renovate/lib/workers/repository/update/branch/index.ts:392:19)\n    at writeUpdates (/usr/src/app/node_modules/renovate/lib/workers/repository/process/write.ts:159:17)\n    at update (/usr/src/app/node_modules/renovate/lib/workers/repository/process/extract-update.ts:151:11)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:61:19)\n    at attributes.repository (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:173:11)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:158:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:17:22"
       }`

Due to this error, only the go.mod file is updated.
If i change renovate version to 33, it's worked without problems. Please explain to me what this error means and how to fix it. Thanks!

**Comment by viceice** [comment]
you probably need to set `binarySource=global` as that's set to install since v33 and somewhere in v34 we converted golang to support install mode.

are youre renovate air-gapped? then set `binarySource=global`

**Reply by viceice** [reply]
- https://github.com/renovatebot/renovate/releases/tag/33.0.0
- https://github.com/renovatebot/renovate/releases/tag/34.36.0

**Reply by XFNeo** [reply]
@viceice Hi! What does air-gapped mean?

**Reply by rarkins** [reply]
Can it reach the internet?

**Reply by XFNeo** [reply]
> Can it reach the internet?

No)

**Reply by viceice** [reply]
that's air-gapped. So you need to use the full renovate image and set `binarySource=global`

**Reply by XFNeo** [reply]
Ok, will try. Thx
